server/admin/admin_server_test.go

```diff
@@ -369,7 +411,25 @@ func TestServer_CreateTree(t *testing.T) {
 
 	ctx := context.Background()
 	for _, test := range tests {
-		setup := setupAdminServer(ctrl, privateKey, false /* snapshot */, test.wantCommit, test.commitErr)
+		sf := keys.NewMockSignerFactory(ctrl)
+		if test.req.GetKeySpec() != nil {
+			if test.req.Tree.GetPrivateKey() != nil || test.req.Tree.GetPublicKey() != nil {
```

Comment:
nil PrivateKey but non-nil PublicKey doesn't cause an error on PEMSignerFactory. Would you mind adding a test case for this scenario on pem_signer_factory_test.go too?